src/tournament_commands/player_commands/decklist.rs

```diff
@@ -29,11 +29,20 @@ async fn decklist(ctx: &Context, msg: &Message, _: Args) -> CommandResult {
      let all_tourns = data.get::<TournamentMapContainer>().unwrap();
      let tourn = all_tourns.get_mut(tourn_id).unwrap();
 
-     if(len(tourn.players[player_name].decklists) == 0) {
+     if(tourn.players[player_name].decks.len == 0) {
          msg.reply(
              &ctx.http,
              "You have not registered any decks for this tournament"
          )
          .await;
      }
+
+     let names = tourn.players[player_name].decks.names;
```

Comment:
In general, it is best not to index directly into a collection unless you've verified that the value is already in the collection. If the given index is not in the collection, the call will panic. Instead, use the `get` method. This returns an `Option` and is much easier to work with.

Reply:
But haven't we already verified that the player is registered for a tournament on line 16. So we could assume that if we are indexing into the collection the player name is already present inside of it?

Reply:
Oh okay that makes sense. I will try using it for this command, and let me know if I have handled it correctly.